projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -104,6 +104,54 @@ public function add_my_home_menu() {
 		$this->update_menu( 'index.php', 'https://wordpress.com/home/' . $this->domain, __( 'My Home', 'jetpack' ), 'manage_options', 'dashicons-admin-home' );
 	}
 
+	/**
+	 * Adds upsell nudge as a menu.
+	 *
+	 * @param object $nudge The $nudge object containing the content, CTA, link and tracks.
+	 */
+	public function add_upsell_nudge( $nudge ) {
+		$message = '
+<div class="upsell_banner">
+	<div class="banner__info">
+		<div class="banner__title">%1$s</div>
+	</div>
+	<div class="banner__action">
+		<button type="button" class="button">%2$s</button>
+	</div>
+</div>';
+
+		$message = sprintf(
+			$message,
+			wp_kses( $nudge['content'], array() ),
+			wp_kses( $nudge['cta'], array() )
+		);
+
+		add_menu_page( 'site-notices', $message, 'manage_options', 'https://wordpress.com' . $nudge['link'], null, null, 1 );
```

Comment:
JITM rules can be filtered by user role (e.g. `->user_is( 'administrator' )`) which makes me think we may have JITMs for non-admins. Should we use another capability for this menu so non-admins can see nudges as well?

Reply:
Current JITMs are only admin specific. But that's a good idea! Let me try it!

Reply:
Since, the permissions logic lies in JITMs I think `read` role is enough?  db35f0c